[Bug]: An error occurred while adding the reporter at path "d"

### Version

28.0.2

### Steps to reproduce

A minimal repo with the bug is available at [ext/jest-28-bug-reporter-default](https://github.com/ext/jest-28-bug-reporter-default):

1. `git clone https://github.com/ext/jest-28-bug-reporter-default.git`
2. `npm install`
3. `npm test`

---

The repo contains a jest preset which sets `reporters: ["default"]` and in this case it references itself with `"preset": "."`.


### Expected behavior

Running `jest` should succeed.

### Actual behavior

> Error: An error occurred while adding the reporter at path "d".
> Cannot find module 'd'

This can also be seen in the logs from [Github actions](https://github.com/ext/jest-28-bug-reporter-default/runs/6217304474?check_suite_focus=true) from the above mentioned repo.

### Additional context

As a workaround modifying `jest-preset.js` as following works but is not the way it is [documented](https://jestjs.io/docs/configuration#reporters-arraymodulename--modulename-options):

```diff
 module.exports = {
-	reporters: ["default"]
+	reporters: [["default"]]
 }
```

It also works if I don't use a preset at all and hardcode `"reporters": ["default"]` directly to `package.json` but that would defeat the purpose of using a preset.

The real preset I have the issue with does of course add additional reporters but as the bug is reproducible without them they are left out.

### Environment

```shell
System:
    OS: Linux 5.10 Gentoo Linux
    CPU: (24) x64 AMD Ryzen 9 3900X 12-Core Processor
  Binaries:
    Node: 16.14.1 - /usr/bin/node
    Yarn: 1.22.10 - /usr/bin/yarn
    npm: 8.6.0 - /usr/bin/npm
  npmPackages:
    jest: ^28.0.2 => 28.0.2
```

--- a/.eslintrc.cjs
+++ b/.eslintrc.cjs
@@ -383,7 +383,7 @@ module.exports = {
     'no-dupe-args': 'error',
     'no-dupe-class-members': 'error',
     'no-dupe-keys': 'error',
-    'no-duplicate-case': 'warn',
+    'no-duplicate-case': 'error',
     'no-else-return': 'off',
     'no-empty': 'off',
     'no-empty-character-class': 'warn',

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 
 ### Fixes
 
+- `[jest-config]` Normalize `reporters` option defined in presets ([#12769](https://github.com/facebook/jest/pull/12769))
 - `[@jest/reporters]` Fix trailing slash in matching `coverageThreshold` key ([#12714](https://github.com/facebook/jest/pull/12714))
 - `[@jest/transform]` Throw better error if an invalid return value if encountered ([#12764](https://github.com/facebook/jest/pull/12764))
 

--- a/packages/jest-config/src/__tests__/normalize.test.ts
+++ b/packages/jest-config/src/__tests__/normalize.test.ts
@@ -1416,6 +1416,55 @@ describe.each(['setupFiles', 'setupFilesAfterEnv'])(
   },
 );
 
+describe("preset with 'reporters' option", () => {
+  beforeEach(() => {
+    const Resolver = require('jest-resolve').default;
+    Resolver.findNodeModule = jest.fn(name => {
+      if (name === 'with-reporters/jest-preset') {
+        return '/node_modules/with-reporters/jest-preset.json';
+      }
+
+      return `/node_modules/${name}`;
+    });
+    jest.doMock(
+      '/node_modules/with-reporters/jest-preset.json',
+      () => ({
+        reporters: ['default'],
+      }),
+      {virtual: true},
+    );
+  });
+
+  afterEach(() => {
+    jest.dontMock('/node_modules/with-reporters/jest-preset.json');
+  });
+
+  test("normalizes 'reporters' option defined in preset", async () => {
+    const {options} = await normalize(
+      {
+        preset: 'with-reporters',
+        rootDir: '/root/',
+      },
+      {} as Config.Argv,
+    );
+
+    expect(options.reporters).toEqual([['default', {}]]);
+  });
+
+  test("overrides 'reporters' option defined in preset", async () => {
+    const {options} = await normalize(
+      {
+        preset: 'with-reporters',
+        reporters: ['summary'],
+        rootDir: '/root/',
+      },
+      {} as Config.Argv,
+    );
+
+    expect(options.reporters).toEqual([['summary', {}]]);
+  });
+});
+
 describe('runner', () => {
   let Resolver;
   beforeEach(() => {

--- a/packages/jest-config/src/normalize.ts
+++ b/packages/jest-config/src/normalize.ts
@@ -942,7 +942,6 @@ export default async function normalize(
       case 'outputFile':
       case 'passWithNoTests':
       case 'replname':
-      case 'reporters':
       case 'resetMocks':
       case 'resetModules':
       case 'restoreMocks':

--- a/packages/jest-config/src/normalize.ts
+++ b/packages/jest-config/src/normalize.ts
@@ -375,14 +375,19 @@ const normalizeRootDir = (
   };
 };
 
-const normalizeReporters = (options: Config.InitialOptionsWithRootDir) => {
-  const reporters = options.reporters;
+const normalizeReporters = ({
+  reporters,
+  rootDir,
+}: Config.InitialOptionsWithRootDir):
+  | Array<Config.ReporterConfig>
+  | undefined => {
   if (!reporters || !Array.isArray(reporters)) {
-    return options;
+    return undefined;
   }
 
   validateReporters(reporters);
-  options.reporters = reporters.map(reporterConfig => {
+
+  return reporters.map(reporterConfig => {
     const normalizedReporterConfig: Config.ReporterConfig =
       typeof reporterConfig === 'string'
         ? // if reporter config is a string, we wrap it in an array
@@ -392,13 +397,13 @@ const normalizeReporters = (options: Config.InitialOptionsWithRootDir) => {
         : reporterConfig;
 
     const reporterPath = replaceRootDirInPath(
-      options.rootDir,
+      rootDir,
       normalizedReporterConfig[0],
     );
 
     if (!['default', 'github-actions', 'summary'].includes(reporterPath)) {
       const reporter = Resolver.findNodeModule(reporterPath, {
-        basedir: options.rootDir,
+        basedir: rootDir,
       });
       if (!reporter) {
         throw new Resolver.ModuleNotFoundError(
@@ -410,8 +415,6 @@ const normalizeReporters = (options: Config.InitialOptionsWithRootDir) => {
     }
     return normalizedReporterConfig;
   });
-
-  return options;
 };
 
 const buildTestPathPattern = (argv: Config.Argv): string => {
@@ -536,12 +539,10 @@ export default async function normalize(
     ],
   });
 
-  let options = normalizeReporters(
-    normalizeMissingOptions(
-      normalizeRootDir(setFromArgv(initialOptions, argv)),
-      configPath,
-      projectIndex,
-    ),
+  let options = normalizeMissingOptions(
+    normalizeRootDir(setFromArgv(initialOptions, argv)),
+    configPath,
+    projectIndex,
   );
 
   if (options.preset) {
@@ -748,6 +749,9 @@ export default async function normalize(
               Array.isArray(transformElement) ? transformElement[1] : {},
             ];
           });
+        break;
+      case 'reporters':
+        value = normalizeReporters(oldOptions);
         break;
       case 'coveragePathIgnorePatterns':
       case 'modulePathIgnorePatterns':

